lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -237,21 +237,34 @@ def add_address(post, options)
       end
 
       def add_invoice(post, money, options)
+        currency = options[:currency] || currency(money)
+        money = what_amount(money, options)
         amount = {
-          value: amount(money),
-          currency: options[:currency] || currency(money)
+          value: localized_amount(money, currency),
+          currency: currency
         }
         post[:amount] = amount
       end
 
       def add_invoice_for_modification(post, money, options)
+        currency = options[:currency] || currency(money)
+        money = what_amount(money, options)
         amount = {
-          value: amount(money),
-          currency: options[:currency] || currency(money)
+          value: localized_amount(money, currency),
+          currency: currency
         }
         post[:modificationAmount] = amount
       end
 
+      # temporary method in place to support Spreedly customers switching over to sending multiplied amounts for non-fractional currency transactions, as now required for localized_amount. To avoid amount manipulation, send opt_out_multiply_amount with any non-fractional currency transaction.
+      def what_amount(money, options)
```

Comment:
Two small nitpicks, though knowing this is a temporary measure, I don't feel strongly about updating these:
1) use a multi-line comment line breaks for the long comment
2) `what_amount` -> `calculate_amount` ??